### PR TITLE
New Custom Modules

### PR DESCRIPTION
**Pull request by wolter**

### CODE REVIEW

File: modules/alexa/README.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
+Adds [Amazon Alexe](https://developer.amazon.com/docs/ask-overviews/build-skills-with-the-alexa-skills-kit.html) specific features to Cognigy.AI
```
Comment (mastasky) [inline modules/alexa/README.md]: Instead of "features", maybe say "Flow Nodes"

File: modules/alexa/package.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	"description": "Alexa Module",
+	"author": "Sascha Wolter",
+	"keywords": [],
+	"main": "build/module.js",
+	"devDependencies": {
+		"@types/node": "^10.12.5",
+		"typescript": "^3.1.6"
+	},
+	"dependencies": {
+		"@types/request-promise": "^4.1.42",
```
Comment (mastasky) [inline modules/alexa/package.json]: this is a dev dependency, please move

File: modules/alexa/src/module.ts [hunk 3]
```diff
@@ -0,0 +1,325 @@
+import * as request_promise from 'request-promise';
+
+function handleError(input: IFlowInput, error:Error, stopOnError:boolean, writeToContext:boolean, store:string): IFlowInput | {}  {
+
```
Comment (mastasky) [inline modules/alexa/src/module.ts]: remove whitespace

File: modules/alexa/src/module.ts [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+    } else {
+        input.input[store] = { "error": error };
+    }
+
+    if (stopOnError) {
+        return Promise.reject(error.message);
+    } else {
+        return input;
+    }
+
```
Comment (mastasky) [inline modules/alexa/src/module.ts]: remove whitespace

File: modules/alexa/src/module.ts [hunk 5]
```diff
@@ -0,0 +1,325 @@
...
+function isAlexa(input: IFlowInput): Boolean  {
+    // Check current channel...
+    return (input.input.channel === "alexa");
+}
+
+/**
+ * Create an "ask for permission consent" card using a list of permissions such as alexa::alerts:reminders:skill:readwrite, read::alexa:device:all:address etc.
+ * 
+ * @arg {CognigyScriptArray} `permissions` Permissions to confirm by the user. See documentation for possible values.
```
Comment (mastasky) [inline modules/alexa/src/module.ts]: Please remove "See documentation for possible values." - this comment will actually show inside the UI

File: modules/alexa/src/module.ts [hunk 6]
```diff
@@ -0,0 +1,325 @@
...
+ */
+async function respondWithPermissionCard(input: IFlowInput, args: {permissions: Array<string>}): Promise<IFlowInput | {}> {
+    let jsonPayload:any = {
+        _cognigy: {
+            _alexa: {
+                response: {
+                    card:  {
+                        type: "AskForPermissionsConsent",
+                        permissions: args.permissions,
```
Comment (mastasky) [inline modules/alexa/src/module.ts]: unused comma

File: modules/alexa/src/module.ts [hunk 7]
```diff
@@ -0,0 +1,325 @@
...
+ * 
+ * @arg {String} `path` Alexa API path to call (something like "/v1/devices/${deviceId}/settings/address")
+ * @arg {JSON} `payload` The payload to send to the Alexa API
+ * @arg {Select[GET,POST,PUT,DELETE]} `method` The REST method to be used
+ * @arg {Boolean} `writeToContext` Whether to write to Cognigy Context (true) or Input (false)
+ * @arg {CognigyScript} `store` Where to store the result
+ * @arg {Boolean} `stopOnError` Whether to stop on error or continue
+ */
+async function callAlexaAPI(input: IFlowInput, args: { path:string, payload:any, method:string, writeToContext: boolean, store: string, stopOnError: boolean }): Promise<IFlowInput | {}> {
+
```
Comment (mastasky) [inline modules/alexa/src/module.ts]: remove whitespace

File: modules/alexa/src/module.ts (hunk 3 above)
Comment (wolter) [reply modules/alexa/src/module.ts]: done

File: modules/alexa/src/module.ts (hunk 4 above)
Comment (wolter) [reply modules/alexa/src/module.ts]: done

File: modules/alexa/src/module.ts (hunk 5 above)
Comment (wolter) [reply modules/alexa/src/module.ts]: done

File: modules/alexa/src/module.ts (hunk 6 above)
Comment (wolter) [reply modules/alexa/src/module.ts]: done

File: modules/alexa/README.md (hunk 1 above)
Comment (wolter) [reply modules/alexa/README.md]: done

Comment (wolter) [reply modules/alexa/README.md]: done

File: modules/alexa/package.json (hunk 2 above)
Comment (wolter) [reply modules/alexa/package.json]: done

**Pull request merged**